gui/src/app/SPAnalysis/loadFilesFromGist.ts

```diff
@@ -0,0 +1,45 @@
+import { Octokit } from "@octokit/rest"
+
+const loadFilesFromGist = async (gistUri: string): Promise<{ files: { [key: string]: string }, description: string }> => {
+    const parts = gistUri.split('/')
+    const gistId = parts[parts.length - 1]
+    const octokit = new Octokit()
+    const r = await octokit.request('GET /gists/{gist_id}', {
+        gist_id: gistId,
+        headers: {
+            'X-GitHub-Api-Version': '2022-11-28'
+        }
+    })
```

Comment:
What does this function do if provided a bad gist ID?

Reply:
It throws an error, which I think is what we want. Error handling on call of fetchRemoteAnalysis could be added.

Reply:
The rest of the calls in fetchRemoteAnalysis currently catch their own errors (e.g. tryFetch)

Reply:
I guess we should decide what behavior we want in the case where a stan, data, project url is not able to be loaded. Do we just not load that part and print to the console? Or is one error sufficient to fail the entire operation and provide an alert?

Reply:
What I did in #39 was populate the text fields with a message, like the stan code would become

`// failed to load from URL_HERE`

But at the moment it is just doing logging

Reply:
I have implemented fallback text like this

```
const stanFilePromise = query.stan ? tryFetch(query.stan, `// Unable to load from ${query.stan}`) : Promise.resolve(undefined);
    const dataFilePromise = query.data ? tryFetch(query.data, `Unable to load from ${query.data}`) : Promise.resolve(undefined);
    const sampling_optsPromise = query.sampling_opts ? tryFetch(query.sampling_opts, undefined) : Promise.resolve(undefined);
```

and I think an exception is appropriate when the entire project does not load.

Does that sound okay?

Reply:
I would prefer something like 

```
    const stanFilePromise = query.stan ? tryFetch(query.stan) : Promise.resolve("");
```

and then have an `else` on the `if (stanFileContent)` following the await. I don't think it is necessary to hand the fallback value to tryFetch just to get it back ourselves.

If you want I can also do this separately

Reply:
Okay I'll let you work on that. But a note... there are two reasons why the stanFileContent might be empty... only one should result in the message in the editor.

Reply:
Yeah, I think the `if` would become an `if (stanFileContent !== undefined)`, with the else then only firing when tryfetch returned undefined.

If you're not interested in doing that here, this is good to merge as far as I am concerned!